cli/src/main.rs

```diff
@@ -307,32 +349,52 @@ fn show_transaction(program: Program, opts: ShowTransactionOpts) {
         println!("    Program data address:    {}", instr.accounts[0].pubkey);
         println!("    Buffer with new program: {}", instr.accounts[2].pubkey);
         println!("    Spill address:           {}", instr.accounts[3].pubkey);
+    } else
+    // Try to deserialize the known multisig instructions. The instruction
+    // data starts with an 8-byte tag derived from the name of the function,
+    // and then the struct data itself, so we need to skip the first 8 bytes
+    // when deserializing. See also `anchor_lang::InstructionData::data()`.
+    // There doesn't appear to be a way to access the tag through code
+    // currently (https://github.com/project-serum/anchor/issues/243), so we
+    // hard-code the tag here (it is stable as long as the namespace and
+    // function name do not change).
+    if instr.program_id == program.id()
+        && &instr.data[..8] == &[122, 49, 168, 177, 231, 28, 167, 204]
+    {
+        if let Ok(instr) =
+            multisig_instruction::SetOwnersAndChangeThreshold::try_from_slice(&instr.data[8..])
+        {
+            println!("  This is a multisig::set_owners_and_change_threshold instruction.");
+            println!(
+                "    New threshold: {} out of {}",
```

Comment:
I just realized, it would be nice to print the diff to see what changed with respect to the old version.